### ISSUE TITLE
[turborepo] Issue using prune + submodules

### ISSUE THREAD

**Issue by mattoni** [vercel/turbo#5485]
### What version of Turborepo are you using?

1.10.7

### What package manager are you using / does the bug impact?

Yarn v2/v3 (node_modules linker only)

### What operating system are you using?

Linux

### Describe the Bug

I'm attempting to use the prune command to focus my workspace for CI/CD + prod release. There are submodules up to a depth of 3 that are important to my app.

`> yarn turbo prune --scope=my-app`
`> cd out`
`> yarn`
`> yarn turbo build --filter=my-app`

This results in an error immediately:

```
 ERROR  run failed: error hashing package files: git error: 'git ls-tree' in /home/-/Dev/-/-/monorepo/out/packages/openapi/public-spec exited with code 128 stderr: fatal: not a git repository: /home/-/Dev/-/-/monorepo/out/packages/openapi/public-spec/../../../.git/modules/packages/openapi/modules/public-spec
```

where both `openapi` and `public-spec` are monorepos, `public-spec` being nested inside `open-api`. 

### Expected Behavior

I had expected this to use non-git based hashing, but I think the submodules are causing issue for one reason or another.

### To Reproduce

Create a new project, and add some submodules, then try to prune + execute a command on the output directory.

**Comment by lyctea** [comment]
> Hi @mattoni, can you try again on `1.10.9`? We've merged #5505, which should fall back to manual file hashing if we encounter any git error and this should play nicer with git submodules.您好，您可以再试一次 `1.10.9` 吗？我们合并了#5505，如果我们遇到任何 git 错误，它应该回退到手动文件哈希，并且这应该与 git 子模块配合得更好。

I upgraded to 1.10.9 and fixed the issue

**Comment by mattoni** [comment]
> Hi @mattoni, can you try again on `1.10.9`? We've merged #5505, which should fall back to manual file hashing if we encounter any git error and this should play nicer with git submodules.

I haven't had time this week, sorry. I will try to test next week. Thanks for addressing this though!

**Comment by chris-olszewski** [comment]
Please reopen if upgrading doesn't solve the issue.
